vault/logical_system_custom_messages.go

```diff
@@ -433,20 +433,30 @@ func (b *SystemBackend) handleCreateCustomMessages(ctx context.Context, req *log
 	}
 
 	var link *uicustommessages.MessageLink
-	if linkMap != nil {
-		link = &uicustommessages.MessageLink{}
-
-		for k, v := range linkMap {
-			href, ok := v.(string)
-			if !ok {
-				return logical.ErrorResponse(fmt.Sprintf("invalid url for %q key in link parameter value", k)), nil
-			}
-
-			link.Title = k
-			link.Href = href
+FORLOOP:
+	for k, v := range linkMap {
```

Comment:
I'm not convinced we doing the processing inside the for loop here, since we're only allowing a single link within the JSON. I think it makes the code a bit more confusing, since putting this inside loop implies to the reader that we're processing multiple things. 

I think it might be a little cleaner to separate the extraction of the values and the validation, which also avoids the need for the special label break, like this:

```
if len(linkMap) != 1 {
   // return error
}
var linkTitle, linkUrl string
for k, v := range linkMap {
  linkTitle = k
  linkUrl = v
  break
}
// do processing with switch statement
```
There might be a cleaner way to extract the 'only' key and value from a 1 length map, but this seemed cleanest to me. It would make things a lot cleaner and easier to understand, I think